src/QoiSharp/QoiEncoder.cs

```diff
@@ -30,9 +36,10 @@ public static byte[] Encode(QoiImage image)
         int height = image.Height;
         byte channels = (byte)image.Channels;
         byte colorSpace = (byte)image.ColorSpace;
-        byte[] pixels = image.Data;
+        ReadOnlySpan<byte> pixels = image.Data.Span;
 
-        byte[] bytes = new byte[QoiCodec.HeaderSize + QoiCodec.Padding.Length + (width * height * channels)];
+        if (pixels.Length < QoiCodec.HeaderSize + QoiCodec.Padding.Length + (width * height * channels))
+            return -1;
```

Comment:
Return -1 if the buffer is too small to encode the image.

Reply:
Great. But then the user needs to be given a method to calculate the minimum buffer size.